lib/note-actions/index.tsx

```diff
@@ -0,0 +1,199 @@
+  pinNote: (noteId: T.EntityId, shouldPin: boolean) => any;
+  publishNote: (noteId: T.EntityId, shouldPublish: boolean) => any;
+  shareNote: () => any;
+  toggleRevisions: () => any;
+  trashNote: () => any;
+};
+
+type Props = StateProps & DispatchProps;
+
+export class NoteActions extends Component<Props> {
+  static displayName = 'NoteActions';
+
+  handleClickOutside = this.props.onOutsideClick;
+
+  getNoteLink = (note: T.Note, noteId: T.EntityId) => {
+    const { title } = getNoteTitleAndPreview(note);
+    return `[${title}](simplenote://note/${noteId})`;
+  };
+
+  getPublishURL = (url: string | undefined) => {
+    return isEmpty(url) ? null : `http://simp.ly/p/${url}`;
+  };
+
+  render() {
+    const { hasRevisions, isMarkdown, isPinned, noteId, note } = this.props;
+    const isPublished = includes(note.systemTags, 'published');
+    const publishURL = this.getPublishURL(note.publishURL);
+    const noteLink = this.getNoteLink(note, noteId);
+
+    return (
+      <div className="note-actions theme-color-bg theme-color-fg theme-color-border">
+        <div className="note-actions-panel theme-color-border">
+          <label
+            className="note-actions-item"
+            htmlFor="note-actions-pin-checkbox"
+          >
+            <span className="note-actions-item-text">
+              <span className="note-actions-name">Pin to top</span>
+            </span>
+            <span className="note-actions-item-control">
+              <CheckboxControl
+                id="note-actions-pin-checkbox"
+                className="theme-color-border"
+                checked={isPinned}
+                onChange={() => {
+                  this.pinNote(!isPinned);
+                }}
+              />
+            </span>
+          </label>
+
+          <label
+            className="note-actions-item"
+            htmlFor="note-actions-markdown-checkbox"
+          >
+            <span className="note-actions-item-text">
+              <span className="note-actions-name">Markdown</span>
+            </span>
+            <span className="note-actions-item-control">
+              <CheckboxControl
+                id="note-actions-markdown-checkbox"
+                checked={isMarkdown}
+                onChange={() => {
+                  this.markdownNote(!isMarkdown);
+                }}
+              />
+            </span>
+          </label>
+
+          <div className="note-actions-item note-actions-internal-link">
+            <div className="note-actions-copy">
+              {/* <p className="note-actions-detail note-actions-link-text theme-color-fg-dim">{`simplenote://note/${noteId}`}</p> */}
+              <ClipboardButton text={noteLink} linkText="Copy Internal Link" />
+            </div>
+          </div>
+
+          {hasRevisions && (
+            <div
+              className="note-actions-item"
+              onClick={this.props.toggleRevisions}
+            >
+              History…
+            </div>
```

Comment:
We should generally avoid using `div` for clickable elements as it removes several accessibility features that elements like `button` and `a` provide (e.g. focusability). What do you think about changing this element to a `button`?

Reply:
I've updated all of these to use buttons now.